packages/@aws-cdk/aws-appmesh/lib/mesh.ts

```diff
@@ -89,10 +89,15 @@ abstract class MeshBase extends cdk.Resource implements IMesh {
   /**
    * Adds a VirtualService with the given id
    */
-  public addVirtualService(id: string, props: VirtualServiceBaseProps = {}): VirtualService {
+  public addVirtualService(
```

Comment:
Sorry, I don't understand these changes 😕.

The whole point of having `addVirtualService()` on Mesh is so that you don't have to provide the Mesh again. But with your changes, you can do something like:

```ts
const mesh = new appmesh.Mesh(...);
mesh.addVirtualService({
  virtualServiceProvider.none(mesh),
});
```

which just doesn't make sense to me.

I would revert all of these changes here, and this method should be:

```suggestion
  public addVirtualService(id: string, props: VirtualServiceBaseProps = {}): VirtualService {
    return new VirtualService(this, id, {
      ...props,
      virtualServiceProvider: VirtualServiceProvider.none(this),
    });
  }
```

Reply:
I guess I was thinking of it more as a utility method, a quick way to add a new Virtual Service to your mesh, similar to how we add other resources like Virtual Nodes. If you cannot specify a provider, then essentially the method is useless as it will never produce a usable Virtual Service. If you created a VS as a placeholder and then went to add a provider you would have to remove it from the `addVirtualService` method into a `new VirtualService(...)` call. How it is here, you could have `addVirtualService('VirtualService')` in one synthesized stack which uses the `none` method as the default. Later, you could come along and add a different provider if you so choose. 

We can remove the method all together if you feel it does not fit well.

Reply:
> I guess I was thinking of it more as a utility method, a quick way to add a new Virtual Service to your mesh

Right, but if you want to specify _any_ property of the created Service, then it doesn't really save anything, right?

```ts
mesh.addVirtualService('Service', {
  virtualServiceName: 'MyService',
  provider: VirtualServiceProvider.none(mesh),
});
```

Doesn't save anything over

```ts
new VirtualService(this, 'Service', {
  virtualServiceName: 'MyService',
  provider: VirtualServiceProvider.none(mesh),
});
```

I think it doesn't make too much sense to over-emphasize the "add a VirtualService to the Mesh with only default properties" use case - it seems to just lead to an awkward API.

> , similar to how we add other resources like Virtual Nodes.

Notice that all other methods int the AppMesh module like `addVirtualNode()`, `addVirtualGateway()`, etc. do _not_ allow you to specify the main owning resource, like the Mesh, again.

I don't think this inconsistency is worth it. I would either not allow passing `virtualServiceProvider` in `addVirtualService()`, or get rid of this method completely, whichever you prefer (if you go with the later, don't forget to add a BREAKING CHANGE note to the PR description).

Reply:
For now, let's remove it. You're right that there is no real difference between the two method calls